### PR TITLE
Fix the bug when external dependencies is used.

### PR DESCRIPTION
**Pull request by chtyim**
- Shouldn't delete the jar directory until the end

### CODE REVIEW

File: src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCompute.java [hunk 1]
```diff
@@ -134,16 +134,15 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
...
     interpreter = context.createSparkInterpreter();
-    File dir = config.getDependencies() == null ? null : Files.createTempDirectory("sparkprogram").toFile();
-    try {
-      if (config.getDependencies() != null) {
-        SparkCompilers.addDependencies(dir, interpreter, config.getDependencies());
-      }
+    if (config.getDependencies() != null) {
+      File tempDir = Files.createTempDirectory("sparkprogram").toFile();
+      SparkCompilers.addDependencies(tempDir, interpreter, config.getDependencies());
+      // TODO: Ideally we need to delete the temp dir when the pipeline exaction finished.
```
Comment (ecfm) [inline src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCompute.java]: Just created this JIRA to be added here: https://issues.cask.co/browse/CDAP-13001

**Comment by ecfm** [comment]
Do we need a new version for this?

**Comment by chtyim** [comment]
Did we ever published 2.0.4?

**Comment by ecfm** [comment]
@chtyim https://github.com/caskdata/cask-marketplace/pull/192 this PR was merged but the build hasn't been wrong. I guess no need to bump version. I can just update the plugin in the market and then run the build

**Pull request merged**